packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -37,10 +40,208 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
-/// Geometry information for scaffold components.
+/// Responsible for placing the [FloatingActionButton] in the [Scaffold].
+/// 
+/// Flutter provides [FabPositioner]s for the common FAB placements in
+/// Material Design apps, whose implementations are available as static
+/// members of this class.
+abstract class FabPositioner {
+  const FabPositioner();
+
+  /// End-aligned [FloatingActionButton], floating at the bottom of the screen.
+  /// 
+  /// This is the default alignment of FABs in Material apps.
+  static const FabPositioner endFloat = const _EndFloatFab();
+
+  /// Centered [FloatingActionButton], floating at the bottom of the screen.
+  static const FabPositioner centerFloat = const _CenterFloatFab();
+
+  /// Places the [FloatingActionButton] based on the [Scaffold]'s layout.
+  Offset getOffset(ScaffoldPrelayoutGeometry scaffoldGeometry);
+
+  @override
+  String toString() => '$runtimeType()';
+}
+
+/// Provider of [FloatingActionButton] animations.
```

Comment:
This needs much more detailed docs. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation for some writing prompts.

Reply:
Done